R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+    dest_file <- paste0(file_name, ".zip")
+    tsf_file <- paste0(file_name, ".tsf")
+  }else{  
+    dest_file <- file.path(destination_folder, paste0(file_name, ".zip"))
+    tsf_file <- file.path(destination_folder, paste0(file_name, ".tsf"))
+  }
+  
+  
+  if(!file.exists(tsf_file)){ # Check whether the required .tsf file is already there at the specified destination folder
+    if(!file.exists(dest_file)){ # Check whether the required .zip file is already there at the specified destination folder. If not download the .zip file from Zenodo
+      tryCatch({
+        download.file(paste0("https://zenodo.org/record/", record_id, "/files/", file_name, ".zip"), destfile = dest_file)
+      }, error = function(e) {   
+        message(e)
+        stop("Record ID or file name is incorrect")
+      })
+    }
+    
+    if(destination_folder != '') # Unzip the downloaded .zip folder
+      unzip(dest_file, exdir = destination_folder)
+    else
+      unzip(dest_file)
+  }
+ 
+  convert_tsf_to_tsibble(tsf_file, value_column_name, key, index) # Creating a tsibble/data.frame depending on the provided key and index
+}
+
+
+
+#' get_forecastingdata function
+#' 
+#' This function downloads a zip file from Zenodo platform given the name of the dataset, unzip the contents and converts the unzipped .tsf file into a tsibble or a data.frame given the key and index.
+#' For more details, please refer to our paper: Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @param dataset Name of the dataset that needs to be downloaded from Zenodo.
+#' @param destination_folder Folder path where the downloaded files need to be stored. By default, it uses the R working directory to store files.
+#' @param index The name of the time attribute that should be used as the index when creating the tsibble. If doesn't provide, it tries to find a valid index within the data. If there is no valid index, then a data frame will be returned instead of a tsibble.
+#' @param key The name of the attribute that should be used as the key when creating the tsibble. The default value is "series_name" which is the key of the datasets in our repository. If doesn't provide, a data frame will be returned instead of a tsibble.
+#' @param value_column_name Any name that is preferred to have as the name of the column containing series values in the returning tsibble.
+#'
+#' @return This function returns a tsibble or a data.frame based on the validity of given key and index values.
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
+get_forecastingdata <- function(dataset=c("nn5", "nn5_without_missing", "nn5_weekly", "m1_yearly", 
```

Comment:
I think this function name could be improved. I do understand the use of 'forecastingdata' as a reference to the website, however I think that 'forecasting' and 'data' should at least be separated. Possibly also mention 'monash' somewhere to be more specific about the source of the data?
`get_monash_forecasting_data()`?

Reply:
Fixed as suggested